nodes/stampzilla-updater/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/stampzilla/stampzilla-go/stampzilla/installer"
+)
+
+const (
+	gitPath = "/home/stampzilla/go/src/github.com/stampzilla/stampzilla-go"
+)
+
+func main() {
+	//installer := installer.NewInstaller()
+	//installer.GoGet("github.com/stampzilla/stampzilla-go/nodes/stampzilla-enocean", true)
+	//return
+
+	wg := &sync.WaitGroup{}
+	quit := make(chan struct{})
+	go signalHandler(wg, quit)
+	go update(wg)
+
+	ticker := time.NewTicker(30 * time.Minute)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				go update(wg)
+			case <-quit:
+				wg.Wait()
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	select {
+	case <-quit:
+		return
+	}
+}
+
+func update(wg *sync.WaitGroup) {
+	wg.Add(1)
+	defer wg.Done()
+
+	if NewCommitExists() {
+		log.Println("New commit exists. Checking if we need to update")
+		installer := installer.NewInstaller()
+		nodes, err := ioutil.ReadDir("/home/stampzilla/go/bin")
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		for _, n := range nodes {
+			if !strings.Contains(n.Name(), "stampzilla-") {
+				continue
+			}
+			repoDate, err := getLatestCommitTimeInFolder(path.Join("nodes", n.Name()))
+			if err != nil {
+				log.Println(err)
+				return
+			}
+
+			if repoDate.After(n.ModTime()) {
+				log.Printf("Update date: %s for node: %s\n", repoDate, n.Name())
+				installer.GoGet("github.com/stampzilla/stampzilla-go/nodes/"+n.Name(), true)
```

Comment:
Stat the binary here. If its newer than n.ModTime AND have a pidfile (running) we should restart it